### PR TITLE
feat: [CCM-15387]: GCP Support Governance

### PR DESCRIPTION
**Pull request by rajarshee-harness**
### Summary

This PR adds Governance Support for GCP Connector (Under a FF)

#### Screenshots

Screenshot 2023-12-19 at 11 58 20 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by rajarshee-harness** [comment]
trigger messagecheck

**Comment by rajarshee-harness** [comment]
trigger messagecheck

**Review by jyotiarora-harness — APPROVED** [review]

File: src/modules/27-platform/connectors/components/CreateConnector/CEGcpConnector/__tests__/__snapshots__/CreateCeGcpConnector.test.tsx.snap
Comment (namanharness) [inline src/modules/27-platform/connectors/components/CreateConnector/CEGcpConnector/__tests__/__snapshots__/CreateCeGcpConnector.test.tsx.snap:1]: we can try to reduce snapshot testing as much as possible.
assertions should work fine.
cc @chetanvekariya-harness

**Review by namanharness — APPROVED** [review]

File: src/modules/27-platform/connectors/components/CreateConnector/CEGcpConnector/__tests__/__snapshots__/CreateCeGcpConnector.test.tsx.snap
Comment (rajarshee-harness) [reply src/modules/27-platform/connectors/components/CreateConnector/CEGcpConnector/__tests__/__snapshots__/CreateCeGcpConnector.test.tsx.snap:1]: Sure. These were existing tests, will create a ticket to refactor tests for CCM connectors.

**Pull request merged**